src/parcsr_ls/par_mgr.c

```diff
@@ -1042,216 +1046,190 @@ hypre_ParCSRMatrixLeftScale(HYPRE_Real *vector,
 /*--------------------------------------------------------------------------
  * hypre_MGRComputeNonGalerkinCoarseGrid
  *
- * Computes the level (grid) operator A_H = RAP, assuming that restriction
- * equals to the injection operator: R = [0 I]
+ * Computes the level (grid) operator A_H = RAP.
  *
  * Available methods:
  *   1: inv(A_FF) approximated by its (block) diagonal inverse
  *   2: CPR-like approx. with inv(A_FF) approx. by its diagonal inverse
  *   3: CPR-like approx. with inv(A_FF) approx. by its block diagonal inverse
  *   4: inv(A_FF) approximated by sparse approximate inverse
+ *   5: Uses classical restriction R = [-Wr I] from input parameters list.
+ *
+ * Methods 1-4 assume that restriction is the injection operator.
+ * Method 5 assumes that interpolation is the injection operator.
  *
  * TODO (VPM): Can we have a single function that works for host and device?
- *             RT is not being used.
  *--------------------------------------------------------------------------*/
 
 HYPRE_Int
-hypre_MGRComputeNonGalerkinCoarseGrid(hypre_ParCSRMatrix    *A,
+hypre_MGRComputeNonGalerkinCoarseGrid(hypre_ParCSRMatrix    *A_FF,
```

Comment:
Why is the function signature changed? Is there a reason one would want to pass in the A_{XX} matrices instead of what was done previously?

Reply:
The signature has changed for performance purposes, i.e., the A_{XX} matrices were being computed inside this function (on master). However, when calling this function, we already know some of them since they are used to build restriction/interpolation. Therefore, we can just pass that information in and we don't need to recompute it inside `hypre_MGRComputeNonGalerkinCoarseGrid`